pkg/storegateway/indexheader/stream_binary_reader.go

```diff
@@ -148,7 +189,26 @@ func newFileStreamBinaryReader(path string, postingOffsetsInMemSampling int, log
 		return nil, err
 	}
 
-	return r, nil
+	return r, err
+}
+
+// writeSamplesToFile uses protocol buffer to write StreamBinaryReader to disk at samplepath.
+func writeSamplesToFile(samplepath string, reader *StreamBinaryReader) error {
+	samples := &indexheaderpb.Samples{}
+
+	samples.Symbols = reader.symbols.NewSymbolSample()
+	samples.PostingsOffsetTable = reader.postingsOffsetTable.NewPostingOffsetTableSample()
+
+	out, err := samples.Marshal()
+	if err != nil {
+		return fmt.Errorf("failed to encode index-header samples: %w", err)
+	}
+
+	if err := os.WriteFile(samplepath, out, 0700); err != nil {
```

Comment:
0700 makes it executable. We should only need read/write. So 0600 should be the correct permissions